packages/webview_flutter/webview_flutter/lib/platform_interface.dart

```diff
@@ -187,6 +188,22 @@ abstract class WebViewPlatformController {
         "WebView loadUrl is not implemented on the current platform");
   }
 
+  /// Loads the URL with postData using "POST" method.
+  ///
+  /// If `url` is not a network URL, it will be loaded with `loadUrl` instead,
+  /// ignoring the `postData` param on Android.
+  ///
+  /// `url` and `postData` must not be null.
+  ///
+  /// Throws an ArgumentError if `url` is not a valid URL string.
+  Future<void> postUrl(
+    String url,
+    Uint8List postData,
```

Comment:
Someone is inevitably going to want to pass headers here.

More generally, anything we might add to one of these methods is likely to apply to the other, which makes me thing we should make a new API that operates in terms of a new simple request class, rather than having a bunch of parameters. That will also mean we could add new things to the requests later without it being a breaking change for implementors, since the request object wouldn't be subclassed by each implementation (unlike method arguments, which are breaking to add).

Reply:
The main reason why we haven't implemented support for headers here is that this is not supported on Android. There is a possible workaround where we do a normal HTTP Post request and load the resulting HTML in the WebView control but this will have the side effect that we need to await for the whole request to be finished before we can load the HTML in the WebView control. Whereas on iOS the WKWebView control can start loading the HTML content directly as it arrives on the response stream. So since there is a difference in behaviour between platforms we choose to stick to the most common implementation (which is indeed a bit limited).

I just wanted to provide this bit of context so you have the full picture on why we choose this direction. I like the idea of adding a new API which operates based on a simple request class. However that also doesn't really solve the issue when dealing with headers on Android. It might even make things worse as it will be more difficult to explain certain fields don't work on certain platforms. 

So maybe good to have a small discussion on this before we pick a certain direction to go on?

Reply:
It's fine to explain that certain features don't work on specific platforms; we do that all the time in plugins. We shouldn't paint ourselves into an API corner here just because we can't currently support a feature on one platform.